src/Views/Filter/Location.cshtml

```diff
@@ -2,16 +2,19 @@
 
 @{
   ViewBag.Title = $"Find courses by location or by training provider";
-  var isInWizard = ViewBag.IsInWizard == true;
+  var isInWizard = ViewBag.IsInWizard;
+  var showBack = !ViewBag.HideBack;
   var selectedRadius = (Model.FilterModel.rad.HasValue) ? Model.FilterModel.rad : (int)RadiusOption.TwentyMiles;
 }
 
-@if (isInWizard) {
-  @Html.Partial("Back", new BackLinkViewModel { Href = Url.Action("Index", "Home"), Title = "Back" })
-}
-else
-{
-  @Html.Partial("Back", new BackLinkViewModel { Href = Url.Action("Index", "Results", Model.FilterModel.ToRouteValues()), Title = "Back to search results" })
+@if (showBack) {
+  @if (isInWizard) {
+    @Html.Partial("Back", new BackLinkViewModel { Href = Url.Action("Index", "Home"), Title = "Back" })
+  }
+  else
+  {
+    @Html.Partial("Back", new BackLinkViewModel { Href = Url.Action("Index", "Results", Model.FilterModel.ToRouteValues()), Title = "Back to search results" })
```

Comment:
This logic doesn't seem right to me. I would have thought that it would only be the home link that is hidden on showBack, not the "back to search results" as well.